signal_scanner_bot/radio_monitor_alert.py

```diff
@@ -32,24 +33,27 @@ def _calculate_lookback_time() -> str:
     return time_stamp_array[0] + time_stamp_array[1][:3]
 
 
-def get_openmhz_calls() -> Dict:
+async def get_openmhz_calls() -> Dict:
     lookback_time = _calculate_lookback_time()
     log.debug(f"Lookback is currently set to: {lookback_time}")
-    response = requests.get(env.OPENMHZ_URL, params={"time": lookback_time})
-    return response.json()["calls"]
+    async with aiohttp.ClientSession(raise_for_status=True) as session:
+        async with session.get(
+            env.OPENMHZ_URL, params={"time": lookback_time}
+        ) as response:
+            return (await response.json())["calls"]
 
 
-def get_pigs(calls: Dict) -> List[Tuple[Dict, str, str]]:
+async def get_pigs(calls: Dict) -> List[Tuple[Dict, str, str]]:
     interesting_pigs = []
     for call in calls:
         time = call["time"]
-        radios = radios = {f"7{radio['src']:0>5}" for radio in call["srcList"]}
+        radios = {"radio": list({f"7{radio['src']:0>5}" for radio in call["srcList"]})}
```

Comment:
Is this right? I recognize that the previous code had two redundant assignments, but I believe that this current code would always at least produce `radios = {"radio": []}` which would be truthy for `len(radios)`. I think this might just need to be:
```suggestion
        radios = {f"7{radio['src']:0>5}" for radio in call["srcList"]}
```

Reply:
I can't remember why I wrapped it in a list, I think something couldn't handle that it was a set or something. Let me look back on it.

Reply:
Yeah something chokes on a set but not a list. I'm updating the len check to do both not `len(radios) or not len(radios) == 0` just to be sure.

Reply:
Oh no that's not right, need to update to `len(radios["radio"])`

Reply:
I'm still not sure why this needs to be a dictionary... it shouldn't be choking on the `len(<set>)` check as that's valid to do in python. Do you know what it was that was causing an issue with the original set?

Reply:
The issue isn't with the len check but the aiohttp library:

```
[2021-11-16 00:24:57,881 - signal_scanner_bot.transport - 164][ERROR] Invalid query type: only str, mapping or sequence of (key, value) pairs is allowed
Traceback (most recent call last):
  File "/app/signal_scanner_bot/transport.py", line 149, in radio_monitor_alert_transport
    radio_monitor_alert_messages := await radio_monitor_alert.check_radio_calls()
  File "/usr/local/lib/python3.8/site-packages/backoff/_async.py", line 133, in retry
    ret = await target(*args, **kwargs)
  File "/app/signal_scanner_bot/radio_monitor_alert.py", line 90, in check_radio_calls
    pigs = await get_pigs(calls)
  File "/app/signal_scanner_bot/radio_monitor_alert.py", line 54, in get_pigs
    async with session.get(env.RADIO_CHASER_URL, params=radios) as response:
  File "/usr/local/lib/python3.8/site-packages/aiohttp/client.py", line 1138, in __aenter__
    self._resp = await self._coro
  File "/usr/local/lib/python3.8/site-packages/aiohttp/client.py", line 507, in _request
    req = self._request_class(
  File "/usr/local/lib/python3.8/site-packages/aiohttp/client_reqrep.py", line 283, in __init__
    url2 = url.with_query(params)
  File "/usr/local/lib/python3.8/site-packages/yarl/_url.py", line 984, in with_query
    new_query = self._get_str_query(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/yarl/_url.py", line 962, in _get_str_query
    raise TypeError(
TypeError: Invalid query type: only str, mapping or sequence of (key, value) pairs is allowed
```

Reply:
Oh yeah I found it. Below link kind of outlines it, basically it expects it to be either `params = [('key', 'value1'), ('key', 'value2')]` or `{'key': ['value1', 'value2']}` for duped parameters and it can't just be like `params=<empty set>` for some reason.

https://docs.aiohttp.org/en/stable/client_quickstart.html#passing-parameters-in-urls

Reply:
ohhhhhh okay that makes total sense. Do you mind adding a comment to the effect of `# Dictionary is required for URL param lists in aiohttp`

Reply:
Yeah good idea.